### PR TITLE
Validate filesystem names

### PR DESCRIPTION
**Pull request by luongthomas**
**Describe the pull request**

Validate filesystem names during creation to avoid bad characters, including FAT special characters.  
Used this as a reference on what characters to filter out:
https://stackoverflow.com/questions/4814040/allowed-characters-in-filename

Screen Shot 2019-03-14 at 11 20 34 AM

Screen Shot 2019-03-14 at 11 20 21 AM

**Link to relevant issues**
N/A

### CODE REVIEW

File: app/src/main/java/tech/ula/utils/ValidationUtility.kt [hunk 1]
```diff
@@ -48,6 +63,19 @@ class ValidationUtility {
         }
     }
 
+    private fun validateFilesystemNameCharacters(filesystemName: String): Boolean {
+        val usernameRegex = "([a-zA-Z0-9!@#$%^&()_+=,.?<>]{0,50})"
```
Comment (MatthewTighe) [inline app/src/main/java/tech/ula/utils/ValidationUtility.kt]: This regex pattern could use a different name imo.

File: app/src/test/java/tech/ula/utils/ValidationUtilityTest.kt [hunk 2]
```diff
@@ -26,6 +26,38 @@ class ValidationUtilityTest {
...
+    @Test
+    fun `Validate fails appropriately if filesystem name is empty`() {
+        val filesystemName = ""
+        credential = validationUtility.validateFilesystemName(filesystemName)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessageId, R.string.error_filesystem_name)
+    }
+
+    @Test
+    fun `Validate fails appropriately if filesystem name is just a period`() {
```
Comment (MatthewTighe) [inline app/src/test/java/tech/ula/utils/ValidationUtilityTest.kt]: Would add a test for the `..` case as well? Or just add it to this test.

File: app/src/main/java/tech/ula/utils/ValidationUtility.kt [hunk 3]
```diff
@@ -48,6 +63,19 @@ class ValidationUtility {
...
+
+        val compiledRegex = Pattern.compile(usernameRegex)
+        val matcher = compiledRegex.matcher(filesystemName)
+        val hasValidCharacters = matcher.matches()
+
+        if (hasValidCharacters) {
```
Comment (MatthewTighe) [inline app/src/main/java/tech/ula/utils/ValidationUtility.kt]: I think you could just `return hasValidCharacters`.

**Comment by luongthomas** [comment]
Added one test and renamed a few variables

**Pull request merged**